presidio-analyzer/presidio_analyzer/nlp_engine/transformers_nlp_engine.py

```diff
@@ -2,153 +2,168 @@
-            )
-            raise KeyError(
-                "Expected keys ('spacy' and 'transformers') was not found in "
-                "models.model_name dict"
-            )
+        super().__init__(models=models, ner_model_configuration=ner_model_configuration)
+        self.entity_key = "bert-base-ner"
 
-        logger.debug(f"Loading SpaCy and transformers models: {models.values()}")
+    def load(self) -> None:
+        """Load the spaCy and transformers models."""
 
+        logger.debug(f"Loading SpaCy and transformers models: {self.models}")
         self.nlp = {}
-        for lang_code, model_name in models.items():
-            nlp = spacy.load(model_name["spacy"], disable=["parser", "ner"])
+
+        for model in self.models:
+            self._validate_model_params(model)
+            spacy_model = model["model_name"]["spacy"]
+            transformers_model = model["model_name"]["transformers"]
+            self._download_spacy_model_if_needed(spacy_model)
+
+            nlp = spacy.load(spacy_model, disable=["parser", "ner"])
             nlp.add_pipe(
-                "transformers",
-                config={"pretrained_model_name_or_path": model_name["transformers"]},
-                last=True,
+                "hf_token_pipe",
+                config={
+                    "model": transformers_model,
+                    "annotate": "spans",
+                    "stride": self.ner_model_configuration.stride,
+                    "alignment_mode": self.ner_model_configuration.alignment_mode,
+                    "aggregation_strategy": self.ner_model_configuration.aggregation_strategy,  # noqa E501
+                    "annotate_spans_key": self.entity_key,
+                },
+            )
+            self.nlp[model["lang_code"]] = nlp
+
+    @staticmethod
+    def _validate_model_params(model: Dict) -> None:
+        if "lang_code" not in model:
+            raise ValueError("lang_code is missing from model configuration")
+        if "model_name" not in model:
+            raise ValueError("model_name is missing from model configuration")
+        if not isinstance(model["model_name"], dict):
+            raise ValueError("model_name must be a dictionary")
+        if "spacy" not in model["model_name"]:
+            raise ValueError("spacy model name is missing from model configuration")
+        if "transformers" not in model["model_name"]:
+            raise ValueError(
+                "transformers model name is missing from model configuration"
             )
-            self.nlp[lang_code] = nlp
+
+    def process_text(self, text: str, language: str) -> NlpArtifacts:
+        """Execute the SpaCy NLP pipeline on the given text and language."""
+        if not self.nlp:
+            raise ValueError("NLP engine is not loaded. Consider calling .load()")
+
+        doc = self.nlp[language](text)
+        return self._doc_to_nlp_artifact(doc, language)
+
+    def _get_entities(self, doc: Doc) -> SpanGroup:
```

Comment:
This part does the filtering, mapping, scoring etc. that was previously in SpacyRecognizer/TransformersRecognizer.